test/framework/resource/k8s/node/wrapper.go

```diff
@@ -15,30 +15,61 @@ package node
 
 import (
 	"context"
+	"fmt"
 
+	cninode "github.com/aws/amazon-vpc-resource-controller-k8s/apis/vpcresources/v1alpha1"
 	"github.com/aws/amazon-vpc-resource-controller-k8s/test/framework/utils"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 )
 
-func GetNodeAndWaitTillCapacityPresent(manager Manager, ctx context.Context, os string, expectedResource string) *v1.NodeList {
-
+func GetNodeAndWaitTillCapacityPresent(manager Manager, os string, expectedResource string) *v1.NodeList {
 	observedNodeList := &v1.NodeList{}
 	var err error
-	err = wait.Poll(utils.PollIntervalShort, utils.ResourceCreationTimeout, func() (bool, error) {
-		By("checking nodes have capacity present")
-		observedNodeList, err = manager.GetNodesWithOS(os)
-		Expect(err).ToNot(HaveOccurred())
-		for _, node := range observedNodeList.Items {
-			_, found := node.Status.Allocatable[v1.ResourceName(expectedResource)]
-			if !found {
-				return false, nil
+	err = wait.PollUntilContextTimeout(context.Background(), utils.PollIntervalShort, utils.ResourceCreationTimeout, true,
+		func(ctx context.Context) (bool, error) {
+			By("checking nodes have capacity present")
+			observedNodeList, err = manager.GetNodesWithOS(os)
+			Expect(err).ToNot(HaveOccurred())
+			for _, node := range observedNodeList.Items {
+				_, found := node.Status.Allocatable[v1.ResourceName(expectedResource)]
+				if !found {
+					return false, nil
+				}
 			}
-		}
-		return true, nil
-	})
+			return true, nil
+		})
 	Expect(err).ToNot(HaveOccurred())
 	return observedNodeList
 }
+
+// VerifyCNINodeCount checks if the number of CNINodes is equal to number of nodes in the cluster, and verifies 1:1 mapping between CNINode and Node objects
+// Returns nil if count and 1:1 mapping exists, else returns error
+func VerifyCNINodeCount(manager Manager) error {
```

Comment:
nit: `VerifyCNINode` is probably better since this func is not just checking count matched.